### PR TITLE
OSDOCS-11891# RWX/RWO SELinux Mount (Dev Preview)

### PR DESCRIPTION
**Pull request by lpettyjo**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-11891

Link to docs preview:

**Main section**: https://81580--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/understanding-persistent-storage.html#storage-persistent-storage-selinux-rwx-rwo_understanding-persistent-storage
**Update to Access Modes table:** https://81580--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/understanding-persistent-storage.html#pv-access-modes_understanding-persistent-storage

QE review:
- [ ] QE has approved this change.

Additional information:

**PTAL**: @jsafrane @gcharot

### CODE REVIEW

File: modules/storage-persistent-storage-pv.adoc [hunk 1]
```diff
@@ -121,23 +121,24 @@ The following table lists the access modes:
...
-|ReadWriteMany
+|ReadWriteMany 
 |`RWX`
 |The volume can be mounted as read-write by many nodes.
 endif::[]
 |===
 --
-1. RWOP uses the SELinux mount feature. This feature is driver dependent, and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, IBM VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
+1. RWO and RWX can use the SELinux mount feature. However, RWO/RWX SELinux Mount is a Developer Preview feature. This feature is not supported by Red Hat, and you should not enable this feature set on production or clusters that you plan to maintain over time. For more information about this feature, see section "RWO/RWX SELinux Mount".
+2. RWOP uses the SELinux mount feature. This feature is driver dependent, and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, IBM VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
```
Comment (DCChadwick) [inline modules/storage-persistent-storage-pv.adoc:141]:
```suggestion
2. RWOP uses the SELinux mount feature. This feature is driver dependent and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, IBM VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
```

**Review by DCChadwick — COMMENTED** [review]
I had a few minor nitpicks. Otherwise, LGTM!

File: modules/storage-persistent-storage-pv.adoc [hunk 2]
```diff
@@ -121,23 +121,24 @@ The following table lists the access modes:
...
 |The volume can be mounted as read-only by many nodes.
-|ReadWriteMany
+|ReadWriteMany 
 |`RWX`
 |The volume can be mounted as read-write by many nodes.
 endif::[]
 |===
 --
-1. RWOP uses the SELinux mount feature. This feature is driver dependent, and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, IBM VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
+1. RWO and RWX can use the SELinux mount feature. However, RWO/RWX SELinux Mount is a Developer Preview feature. This feature is not supported by Red Hat, and you should not enable this feature set on production or clusters that you plan to maintain over time. For more information about this feature, see section "RWO/RWX SELinux Mount".
```
Comment (jsafrane) [inline modules/storage-persistent-storage-pv.adoc:140]: I am not sure we need to mention SELinux for RWO and RWX. I would keep only RWOP in this page for now.

File: storage/understanding-persistent-storage.adoc [hunk 3]
```diff
@@ -16,6 +16,10 @@ include::modules/storage-persistent-storage-reclaim.adoc[leveloffset=+2]
 
 include::modules/storage-persistent-storage-pv.adoc[leveloffset=+1]
 
+include::modules/storage-persistent-storage-selinux-rwx-rwo.adoc[leveloffset=+2]
+
+For more information about RWX/RWO SELinux Mount feature, see ???add xref???
```
[the remark this answers is not in the document]
Comment (lpettyjo) [reply storage/understanding-persistent-storage.adoc:21]: Added, but want to check to see that it is final.

File: modules/storage-persistent-storage-pv.adoc (hunk 2 above)
Comment (lpettyjo) [reply modules/storage-persistent-storage-pv.adoc:140]: @gcharot Do you agree? No problem if you guys want this removed from here.

Comment (lpettyjo) [reply modules/storage-persistent-storage-pv.adoc]: @jsafrane @gcharot OK. But we still want to keep the main doc section about this feature (see https://81580--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/understanding-persistent-storage.html#storage-persistent-storage-selinux-rwx-rwo_understanding-persistent-storage) and the RN notice, correct?

Comment (gcharot) [reply modules/storage-persistent-storage-pv.adoc]: I don't think we need to refer to it in the proper docs, RN is enough

File: modules/storage-persistent-storage-pv.adoc [hunk 4]
```diff
@@ -121,23 +121,24 @@ The following table lists the access modes:
...
-|ReadWriteMany
+|ReadWriteMany 
 |`RWX`
 |The volume can be mounted as read-write by many nodes.
 endif::[]
 |===
 --
-1. RWOP uses the SELinux mount feature. This feature is driver dependent, and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, IBM VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
+1. RWX and RWO can use the SELinux mount feature. However, RWX/RWO SELinux Mount is a Developer Preview feature. This feature is not supported by Red Hat, and you should not enable this feature set on production or clusters that you plan to maintain over time. For more information about this feature, see section "RWX/RWO  SELinux Mount".
+2. RWOP uses the SELinux mount feature. This feature is driver dependent, and enabled by default in ODF, AWS EBS, Azure Disk, GCP PD, {ibm-title} VPC Block, Cinder, and vSphere. For third-party drivers, please contact your storage vendor.
```
[the remark this answers is not in the document]
Comment (lpettyjo) [reply modules/storage-persistent-storage-pv.adoc:141]: @jsafrane @gcharot OK. But we still want to keep the [main doc reference](https://81580--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/understanding-persistent-storage.html#storage-persistent-storage-selinux-rwx-rwo_understanding-persistent-storage) and the RN notice, right?

File: modules/storage-persistent-storage-pv.adoc (hunk 2 above)
Comment (lpettyjo) [reply modules/storage-persistent-storage-pv.adoc]: Closing this PR then, since RN content is in separate PR.

**Comment by lpettyjo** [comment]
Closing this Dev Preview feature, since we want no doc mention. Only reference will be in RNs. https://github.com/openshift/openshift-docs/pull/81552

**Pull request closed without merging**